tests/e2e/mnist_raycluster_sdk_oauth_test.py

```diff
@@ -0,0 +1,158 @@
+import requests
+
+from time import sleep
+
+from torchx.specs.api import AppState, is_terminal
+
+from codeflare_sdk.cluster.cluster import Cluster, ClusterConfiguration
+from codeflare_sdk.job.jobs import DDPJobDefinition
+
+import pytest
+
+from support import *
+
+# This test Creates a Ray cluster with openshift_oauth enable and covers the Ray Job submission with authentication and without authentication functionality
+
+
+@pytest.mark.openshift
+class TestRayClusterSDKOauth:
+    def setup_method(self):
+        initialize_kubernetes_client(self)
+
+    def teardown_method(self):
+        delete_namespace(self)
+
+    def test_mnist_ray_cluster_sdk_auth(self):
+        self.setup_method()
+        create_namespace(self)
+        self.run_mnist_raycluster_sdk_oauth()
+
+    def run_mnist_raycluster_sdk_oauth(self):
+        ray_image = get_ray_image()
+
+        cluster = Cluster(
+            ClusterConfiguration(
+                name="mnist",
+                namespace=self.namespace,
+                num_workers=1,
+                head_cpus="500m",
+                head_memory=2,
+                min_cpus="500m",
+                max_cpus=1,
+                min_memory=1,
+                max_memory=2,
+                num_gpus=0,
+                instascale=False,
+                image=ray_image,
+                openshift_oauth=True,
+            )
+        )
+
+        cluster.up()
+        self.assert_appwrapper_exists()
+
+        cluster.status()
+
+        cluster.wait_ready()
+        self.assert_raycluster_exists()
```

Comment:
Is calling this function redundant? 
If `cluster.wait_ready()` passes it is safe to assume the Ray Cluster was created.

Reply:
I noticed we were doing similar check in [sdk test ](https://github.com/Srihari1192/codeflare-sdk/blob/361a9ec461f1394b7b2fcb32e41b689b265e0749/tests/e2e/mnist_raycluster_sdk_test.py#L90), so i kept same here aswell

Reply:
In assertion we are calling ray API call to check resource is created or not , not sure we are performing same check  in cluster.wait_ready()

Reply:
I suppose its okay to keep its just that `wait_ready` would fail if a ray cluster didn't exist. :+1:

Reply:
@Bobbins228  yes it makes sense to skip this check, updated the code